libraries/chain/include/eosio/chain/host_context.hpp

```diff
@@ -554,12 +554,12 @@ class host_context {
 public:
    vector<account_name> get_active_producers() const;
 
-   virtual int get_action( uint32_t type, uint32_t index, char* buffer, size_t buffer_size )const = 0;
-   virtual int get_context_free_data( uint32_t index, char* buffer, size_t buffer_size )const = 0;
+   virtual int get_action( uint32_t type, uint32_t index, char* buffer, size_t buffer_size ) const { EOS_ASSERT(false, unaccessible_api, "default implemention should never be used"); }
+   virtual int get_context_free_data( uint32_t index, char* buffer, size_t buffer_size )const { EOS_ASSERT(false, unaccessible_api, "default implemention should never be used"); }
```

Comment:
well, I guess for these couldn't just do an `assert()`, but would also need a `__builtin_unreachable()` to silence the warning of not having a return value

Reply:
https://github.com/AntelopeIO/spring/pull/1296/commits/05b812af54d98d0f7cb5e537adc106f1412fc6d9